util/reggen/chip.py

```diff
@@ -0,0 +1,79 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+'''Code representing the entire chip for reggen'''
+
+from typing import Dict, List, Tuple, Union
+
+from .block_base import BlockBase
+from .block import Block
+from .params import Params
+from .reg_block import RegBlock
+from .window import Window
+
+_Triple = Tuple[int, str, Block]
+
+
+class Chip(BlockBase):
```

Comment:
In OpenTitan, the thing represented by this class is typically called a Toplevel. Not that this name is necessarily better, but it's much more common (e.g. "topgen" vs "chipgen", "top_earlgrey" vs "chip_earlgrey" etc). I'd propose renaming this class.

Reply:
Cool, I've gone for `Top` since it's shorter than `Toplevel` and also hasn't got the ambiguous case problem (`TopLevel`?).